Source/WebCore/html/TextFieldInputType.cpp

```diff
@@ -594,7 +594,10 @@ static bool isAutoFillButtonTypeChanged(const AtomString& attribute, AutoFillBut
 
 String TextFieldInputType::sanitizeValue(const String& proposedValue) const
 {
-    return limitLength(proposedValue.removeCharacters(isHTMLLineBreak), HTMLInputElement::maxEffectiveLength);
+    auto proposedValueWithoutLineBreaks = proposedValue.removeCharacters([](auto character) {
+        return isHTMLLineBreak(character);
+    });
```

Comment:
Exciting that this helps so much. Maybe we can change the design so it’s not tempting to do it the less efficient way.

Can this be `auto` instead of `auto&`?

Reply:
I am already using `auto` already?

Reply:
I think Darin is referring to the reference in Source/WebCore/html/EmailInputType.cpp

Reply:
Oh, indeed. I'll fix, thanks.

Reply:
> Maybe we can change the design so it’s not tempting to do it the less efficient way.

I agree. I was very surprised to see `isHTMLLineBreak` in Speedometer profiles, I would have expected it to get inlined properly.